### PR TITLE
Dynmap Chat Integration

### LINKED ISSUE #66: Dynmap Chat Integration
see https://github.com/webbukkit/dynmap/blob/v3.0/DynmapCoreAPI/src/main/java/org/dynmap/DynmapCommonAPI.java#L28 for API usage

### PR DESCRIPTION
**Pull request by EOT3000**
Fixes #66 
CaptureDynmapDiscord
CaptureDynmap

### CODE REVIEW

File: src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.plugin.Plugin;
+import org.dynmap.DynmapCommonAPI;
+import org.dynmap.DynmapWebChatEvent;
+
+public class DynmapHook implements OtherSoftwareHook {
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void dynmapChatEvent(DynmapWebChatEvent event) {
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java:15]: onDynmapChatEvent

File: src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        DiscordUtil.sendMessage(DiscordSRV.getPlugin().getMainTextChannel(), "**[Dynmap]** " + event.getName() + "» " + event.getMessage());
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java:16]: Lacks customizability

File: src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @Override
+    public Plugin getPlugin() {
+        return PluginUtil.getPlugin("dynmap");
+    }
+
+    @Override
+    public void broadcast(String message) {
+        DynmapCommonAPI api = (DynmapCommonAPI) getPlugin();
+
+        api.sendBroadcastToWeb("", message);
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java:28]: First parameter is the sender name, that should be the Discord user/nicknane

File: src/main/java/github/scarsz/discordsrv/hooks/othersoftware/OtherSoftwareHook.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package github.scarsz.discordsrv.hooks.othersoftware;
+
+import github.scarsz.discordsrv.hooks.PluginHook;
+
+//For other message receiving software such as dynmap
+//Not sure if you want to make other non plugin hooks, for things in the config
+public interface OtherSoftwareHook extends PluginHook {
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/hooks/othersoftware/OtherSoftwareHook.java:7]: I'm not sure how I feel sbout this, but the comments gotta go for sure

File: pom.xml [hunk 5]
```diff
@@ -276,6 +276,10 @@
             <id>vankka</id>
             <url>https://nexus.vankka.dev/repository/maven-public/</url>
         </repository>
+        <repository>
+            <id>dynmap</id>
+            <url>http://repo.mikeprimm.com/</url>
```
Comment (Vankka) [inline pom.xml:281]: Probably should be added to / proxied with nexus.scarsz.me @Scarsz (I'm not able to proxy repos/opinion)

File: src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java (hunk 3 above)
Comment (EOT3000) [reply src/main/java/github/scarsz/discordsrv/hooks/othersoftware/DynmapHook.java:28]: the message contains the word "Discord", the role and the name, so I purposefully left the sender blank

**Comment by Vankka** [comment]
Added, made some adjustments in the above commit

**Pull request closed without merging**